goblet/resources/jobs.py

```diff
@@ -65,57 +64,49 @@ def _deploy(self, source=None, entrypoint=None, config={}):
             ]
 
             job_spec = {
-                "apiVersion": "run.googleapis.com/v1",
-                "kind": "Job",
-                "metadata": {
-                    "name": job_name,
-                    "annotations": {"run.googleapis.com/launch-stage": "BETA"},
-                    "labels": gconfig.labels,
-                },
-                "spec": {
+                "launchStage": "BETA",
+                "labels": gconfig.labels,
+                "template": {
+                    "taskCount": len(job.keys()) - 1,
                     "template": {
-                        "metadata": {"annotations": annotations},
-                        "spec": {
-                            "taskCount": len(job.keys()) - 1,
-                            "template": {
-                                "spec": {
-                                    "containers": [container],
-                                    **(gconfig.job_spec or {}),
-                                }
-                            },
-                        },
-                        **job["execution_spec"],
-                    }
+                        "containers": [container],
+                        **(gconfig.job_spec or {}),
+                    },
+                    **job["execution_spec"],
                 },
             }
 
             self.deploy_job(job_name, job_spec)
+            self.set_iam_policy(job_name, gconfig)
```

Comment:
do we want to run this every time, or only for a scheduled job?
should we add a flag `--set-iam` so its optional for users. 
that way the deploy wont fail if the user doesnt have permission to set iam?

Reply:
only when a scheduled job is deployed would make more sense, i'll get that updated.

i can also add the flag so it does not run automatically, or maybe log the failure if it's a permissions issue but continue deploying?